### DISCUSSION TITLE
Concurrent connections to multiple sessions

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8178] [Q&A] [answered]
Hello, I'm creating a function that allows you to use RDP functionality without installation on the web.

I'm making it by referring to the Android version (not the Android environment)

Is there a problem with simultaneous access to multiple sessions using a headless sample client?

Loading dynamic libraries built using JNI and logging in multiple sessions at the same time seems to consume too much memory and resources. ( OOM problem )

Structurally, can multiple sessions be accessed simultaneously in this way cause problems?

Create one context per session and separate it by the instance value.

**Comment by akallabeth** [comment, marked as answer]
* It can be done (and has been done), just not using `JNI`.
* Memory usage depends a lot on your resolution, features (sound? microphone? ...)

**Reply by kimcoldwater** [reply]
Thank you for your answer.

Have you accessed multiple sessions simultaneously on the Android version?
